modules/programs/hyprpaper.nix

```diff
@@ -0,0 +1,108 @@
+{ pkgs, config, lib, ... }:
+
+with lib;
+
+let cfg = config.programs.hyprpaper;
+in {
+  meta.maintainers = [ lib.maintainers.silmarp ];
+
+  options = {
+    programs.hyprpaper = {
+      enable = mkEnableOption "hyprpaper";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.hyprpaper;
+        defaultText = literalExpression "pkgs.hyprpaper";
+        description = "The package to use for the hyprpaper binary.";
+      };
+
+      systemd = {
```

Comment:
I think this is best put in a separate module `services.hyprpaper`. We have a few such modules, e.g., emacs and have made such a separation there.